x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/DateUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.util;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+
+public class DateUtils {
+
+    // TODO: do we have a java.time based parser we can use instead?
+    private static final DateTimeFormatter UTC_DATE_FORMATTER = ISODateTimeFormat.dateOptionalTimeParser().withZoneUTC();
+
+    public static ZoneId UTC = ZoneId.of("UTC");
+
+    private DateUtils() {}
+
+
+    public static ZonedDateTime of(long millis) {
```

Comment:
Long is the milliseconds since epoch in UTC. Mentioning the timezone is redundant and confusing as it implies there might be a long with a different timezone then UTC which is not the case.